core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerAPI.java

```diff
@@ -319,6 +326,40 @@ private TableSequencerEntry openSequencerLocked(CharSequence tableName, Sequence
         return getTableSequencerEntry(Chars.toString(tableName), lock, this.openSequencerInstanceLambda);
     }
 
+    private void readWalTableInfo(
+            CharSequence tableName,
+            WalTableInfo tableInfo,
+            LastWalTxnReader lastTxnReader,
+            Path path,
+            int rootLen
+    ) {
+        tableInfo.reset();
+        // Fast path.
+        // We don't need to allocate a String for the table name here since we don't create a new map entry.
+        seqRegistry.compute(tableName, (name, sequencer) -> {
```

Comment:
Lambda creates garbage

Reply:
Yes, but it's not possible to get rid of it since it's a capturing lambda.

Reply:
We don't need to compute here, `get` will be fine for fast path

Reply:
`get` won't have any atomicity guarantees: we may get a `null` from `get`, start working with it, sleep/get kicked off the game by the OS scheduler while a sequencer would get created and handle a lot of operations. I kept the original behavior from the atomicity perspective as is.

Reply:
Gave the non-atomic reloading a go in 13f96397587129b0d497d069d0bd8db529a62ae7

I don't see any problems with loosing atomicity with the current code base, but that's something we should keep in mind in future as it seems fragile.